### PR TITLE
fix #791

### LINKED ISSUE #791: Static `@property` function wrongly marked as needing `const`, `inout`, or `immutable`
```d
private class Foo {
	static int bar() @property {
		return 2;
	}
}
```
D-scanner suggests putting a `const`, `inout`, or `immutable` attribute on `bar`; however, doing this would result in a compilation error.

### PR DESCRIPTION
**Pull request by brianush1**

### CODE REVIEW

**Review by WebFreak001 — CHANGES_REQUESTED** [review]
this doesn't seem to change the `package` / `package(xyz)` behavior referenced in #880, but only disables this check inside aggregates. (which seems wrong)

Also this is missing a unittest to check the code from the referenced issue.

**Comment by brianush1** [comment]
different issue, this PR is for an issue I opened in 2020

**Comment by WebFreak001** [comment]
ah sorry, missed that. This is still missing a test for the code from that issue though.

**Comment by brianush1** [comment]
just added tests, not sure if github notifies for that so commenting this just in case

File: src/dscanner/analysis/function_attributes.d [hunk 1]
```diff
@@ -147,3 +148,39 @@ private:
...
+			int barConst() const @property { return 0; }
+		}
+
+		interface InterfaceName {
+			int bar() @property; // [warn]: Zero-parameter '@property' function should be marked 'const', 'inout', or 'immutable'.
+			int barConst() const @property;
+
+			abstract int method(); // [warn]: 'abstract' attribute is redundant in interface declarations
+		}
+	}c, sac);
```
Comment (WebFreak001) [inline src/dscanner/analysis/function_attributes.d:183]: can you also test static methods? (as mentioned in #791)

Comment (brianush1) [reply src/dscanner/analysis/function_attributes.d:201]: done

**Review by WebFreak001 — APPROVED** [review]

**Pull request merged**